lua/orgmode/org/indent.lua

```diff
@@ -92,10 +92,17 @@ end
 
 local function foldtext()
   local line = vim.fn.getline(vim.v.foldstart)
+
   if config.org_hide_leading_stars then
-    return vim.fn.substitute(line, '\\(^\\*\\+\\)', '\\=repeat(" ", len(submatch(0))-1) . "*"', '')
-      .. config.org_ellipsis
+    line = vim.fn.substitute(line, '\\(^\\*\\+\\)', '\\=repeat(" ", len(submatch(0))-1) . "*"', '')
+  end
+
+  if vim.opt.conceallevel:get() > 0 then
+    line = string.gsub(line, "%[%[.-%]%[(.-)%]%]", function(captures)
```

Comment:
This does not seem to work with folded `[[links]]` that have no name. I hacked it with 
```
line = string.gsub(line, "%[%[(.-)%]%[?(.-)%]?%]", function(link, name)
    if (name == "") then
        return link
    else
        return name
    end  
end)
```